// @pulumi/github/config/vars.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
const __config = new pulumi.Config("github");
Object.defineProperty(exports, "appAuth", {
    get() {
        return __config.getObject("appAuth");
    },
    enumerable: true,
});
Object.defineProperty(exports, "baseUrl", {
    get() {
        var _a;
        return (_a = __config.get("baseUrl")) !== null && _a !== void 0 ? _a : (utilities.getEnv("GITHUB_BASE_URL") || "https://api.github.com/");
    },
    enumerable: true,
});
Object.defineProperty(exports, "insecure", {
    get() {
        return __config.getObject("insecure");
    },
    enumerable: true,
});
Object.defineProperty(exports, "maxRetries", {
    get() {
        return __config.getObject("maxRetries");
    },
    enumerable: true,
});
Object.defineProperty(exports, "organization", {
    get() {
        return __config.get("organization");
    },
    enumerable: true,
});
Object.defineProperty(exports, "owner", {
    get() {
        return __config.get("owner");
    },
    enumerable: true,
});
Object.defineProperty(exports, "parallelRequests", {
    get() {
        return __config.getObject("parallelRequests");
    },
    enumerable: true,
});
Object.defineProperty(exports, "readDelayMs", {
    get() {
        return __config.getObject("readDelayMs");
    },
    enumerable: true,
});
Object.defineProperty(exports, "retryDelayMs", {
    get() {
        return __config.getObject("retryDelayMs");
    },
    enumerable: true,
});
Object.defineProperty(exports, "retryableErrors", {
    get() {
        return __config.getObject("retryableErrors");
    },
    enumerable: true,
});
Object.defineProperty(exports, "token", {
    get() {
        var _a;
        return (_a = __config.get("token")) !== null && _a !== void 0 ? _a : utilities.getEnv("GITHUB_TOKEN");
    },
    enumerable: true,
});
Object.defineProperty(exports, "writeDelayMs", {
    get() {
        return __config.getObject("writeDelayMs");
    },
    enumerable: true,
});
